dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilder.java

```diff
@@ -82,22 +80,11 @@ ClassPathResult resolveWithMaven(Artifact rootArtifact) {
   }
 
   private ClassPathResult mediate(DependencyGraph result) {
-    // TODO should DependencyGraphResult have a mediate() method that returns a ClassPathResult?
-
     // To remove duplicates on (groupId:artifactId) for dependency mediation
     MavenDependencyMediation mediation = new MavenDependencyMediation();
 
-    LinkedListMultimap<ClassPathEntry, DependencyPath> multimap = LinkedListMultimap.create();
-    List<DependencyPath> dependencyPaths = result.list();
-    for (DependencyPath dependencyPath : dependencyPaths) {
-      Artifact artifact = dependencyPath.getLeaf();
-      mediation.put(dependencyPath);
-      if (mediation.selects(artifact)) {
```

Comment:
This has moved to `MavenDependencyMediation`. The logic of calling selects and put are too Maven specific; Gradle needs to select artifacts after visiting all nodes in the graph.